spec/money_spec.rb

```diff
@@ -374,16 +374,8 @@
       expect((0.5 <=> money)).to eq(-1)
     end
 
-    it "raises error if compared other is not compatible" do
-      expect{ Money.new(5.00) <=> nil }.to raise_error(TypeError)
-    end
-
-    it "have the same hash value as $1" do
-      expect(money.hash).to eq(Money.new(1.00).hash)
-    end
-
-    it "does not have the same hash value as $2" do
-      expect(money.hash).to eq(Money.new(1.00).hash)
+    it "<=> returns nil with non-coercible non-money objects" do
+      expect((money <=> nil)).to(eq(nil))
```

Comment:
the current behaviour is
```ruby
[1] pry(main)> Money.new(1) == nil
=> false
[2] pry(main)> Money.new(1) > nil
TypeError: NilClass can't be coerced into Money
from /Users/michaelelfassy/.gem/ruby/2.5.5/gems/shopify-money-0.11.9/lib/money/money.rb:413:in `arithmetic'
```
Are you saying that it makes more sense to have
```
[1] pry(main)> Money.new(1) > nil 
=> nil
```

Reply:
When `<=>` returns nil, if you do `>`, `>=`, `<`, `<=` it'll still raise an exception. BUT `==` returns false

Reply:
And particularly, we want `<=>` to itself never raise an exception (by convention)

Reply:
not sure about the convention, money matches the behaviour of integers:
```
[3] pry(main)> 1 > nil
ArgumentError: comparison of Integer with nil failed
from (pry):3:in `>'
[4] pry(main)> 1 == nil
=> false
```

Reply:
We raise a `TypeError` with a custom error message instead of the standard `ArgumentError`.

That said, actual motivation is related to aspects of certain deprecated behaviour with `MoneyBag` which delegates down to money, and some _specific_ uses of `<=>` directly (vs `==`). This fix moves us _closer_ to `<=>` behaving consistently with `==` (i.e. `<=>` returning `nil` when `==` returns false, instead of `<=>` raising an exception when `==` returns false).

I can add some extra tests before making this change, so we can showcase what the change in behaviour is if that'll help raise confidence?